frontend/src/App.js

```diff
@@ -1,26 +1,29 @@
 import React, { Component } from 'react';
-import { Switch, Route } from 'react-router-dom';
+import { Switch, Route, withRouter } from 'react-router-dom';
 import Homepage from './containers/Homepage/Homepage.js';
 import LoginPage from './containers/LoginPage/LoginPage.js';
 import SignUp from './components/SignUp/SignUp.js';
 import UsersAll from './components/AdminControls/UsersAll.js';
 import User from './components/AdminControls/User.js';
-import history from './history.js';
+// import history from './history.js';
  
 class App extends Component {
+
+
   render() {
+
     return (
       <div>
         <Switch>
           <Route exact path='/' component={Homepage}></Route>
           <Route path='/login' component={LoginPage}></Route>
           <Route path='/signup' component={SignUp}></Route>
-          <Route exact path='/users' history={history} render={(props) => <UsersAll {...props} />}></Route>
-          <Route exact path='/users/:id' history={history} render={(props) => <User {...props} />}></Route>
+          <Route exact path='/users' render={(props) => <UsersAll {...props} />}></Route>
+          <Route exact path='/users/:id' render={(props) => <User {...props} />}></Route>
         </Switch>
       </div>
     )
   }
 }
 
-export default App;
\ No newline at end of file
+export default withRouter(App);
```

Comment:
Based on my understanding, I don't think we need this. The components that are linked up through `Route`s should automatically have access to the router stuff. Any of the sub-components where we need to do router shenanigans, we can use `withRouter` to make them available.